src/components/global-toolbar/global-toolbar.test.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import ConnectedGlobalToolbar, {
+  GlobalToolbar,
+  mapStateToProps,
+  mapDispatchToProps,
+} from './index';
+import { mockState, setup } from '../../utils/state.mock';
+
+describe('GlobalToolbar', () => {
+  it('renders without crashing', () => {
+    const wrapper = setup.mount(<ConnectedGlobalToolbar />);
+    expect(wrapper.find('.pipeline-icon-toolbar__button').length).toBe(5);
+  });
+
+  it('hides all buttons (except logo and route buttons) when visible prop is false for each of them', () => {
+    const visible = {
+      themeBtn: false,
+      settingsBtn: false,
+    };
+    const wrapper = setup.mount(<ConnectedGlobalToolbar />, { visible });
+    expect(wrapper.find('.pipeline-icon-toolbar__button').length).toBe(3);
+  });
+
+  it('hides one button when visible prop is false for one of them', () => {
+    const visible = {
+      themeBtn: false,
+    };
+    const wrapper = setup.mount(<ConnectedGlobalToolbar />, { visible });
+    expect(wrapper.find('.pipeline-icon-toolbar__button').length).toBe(4);
+  });
+
+  const functionCalls = [
+    ['.pipeline-menu-button--theme', 'onToggleTheme'],
+    ['.pipeline-menu-button--settings', 'onToggleSettingsModal'],
+  ];
+
+  test.each(functionCalls)(
+    'calls %s function on %s button click',
+    (selector, callback) => {
+      const mockFn = jest.fn();
+      const props = {
+        theme: mockState.animals.theme,
+        visible: mockState.animals.visible,
+        [callback]: mockFn,
+      };
+      const wrapper = setup.mount(<GlobalToolbar {...props} />);
+      expect(mockFn.mock.calls.length).toBe(0);
+      wrapper.find(selector).find('button').simulate('click');
+      expect(mockFn.mock.calls.length).toBe(1);
+    }
+  );
+
+  it('maps state to props', () => {
+    const expectedResult = {
+      theme: expect.stringMatching(/light|dark/),
+      visible: expect.objectContaining({
+        exportBtn: expect.any(Boolean),
+        settingsBtn: expect.any(Boolean),
```

Comment:
As per my previous comment, I don't think we need to import the `settingsBtn` and `themeBtn` here as there isn't any use case for them to be false